src/main/java/seedu/address/model/Model.java

```diff
@@ -99,6 +99,11 @@ public interface Model {
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
 
+    /**
+     * Gets predicate for the people in the current filtered list.
+     */
+    Predicate<? super Person> getFilteredPersonListPredicate();
```

Comment:
Maybe make this method return Optional<Predicate<? super Person>> instead of wrapping it in CleanCommand?